### PR TITLE
[PLGN-730] OKTA task backfill updates

### PR DESCRIPTION
**Pull request by ablakley-r7**
### Description

Describe the proposed changes:

  - Update plugin to use SDK 5.4
  - Update to use custom_config in monitor_logs task
  - Update to gather filter_hours(last_24_hours) from custom_config

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

**Comment by ablakley-r7** [comment]
PluginValidator failed due to manual schema change. Automated refresh is causing an update that results in a invalid schema for task monitor_logs. Will investigate seperately

`  "items": {
    "type": "object"
  },`
  
  ->
  
  `  "items": {
    "$ref": {}
  },`

**Review by igorski-r7 — APPROVED** [review]

File: plugins/okta/help.md [hunk 1]
```diff
@@ -1612,6 +1612,7 @@ by Okta themselves, or constructed by the plugin based on the information it has
 
 # Version History
 
+* 4.2.4 - Monitor Logs tasK: Update to latest SDK which adds new task custom_config parameter | Update validators to 0.22.0
```
Comment (dlaverty-r7) [inline plugins/okta/help.md:1615]: (Minor) K in task should be lower case.

**Review by dlaverty-r7 — APPROVED** [review]

**Review by cmcnally-r7 — APPROVED** [review]

File: plugins/okta/komand_okta/tasks/monitor_logs/task.py [hunk 2]
```diff
@@ -170,3 +178,16 @@ def get_last_collection_timestamp(self, new_logs: list, state: dict) -> str:
             new_ts = state_time
 
         return new_ts
+
+    def _get_filter_time(self, custom_config: Dict[str, int]) -> int:
+        """
+        Apply custom_config params (if provided) to the task. If a lookback value exists, it should take
+        precedence (this can allow a larger filter time), otherwise use the default value.
+        :param custom_config: dictionary passed containing `default` or `lookback` values
+        :return: filter time to be applied in `_filter_and_sort_recent_events`
```
Comment (joneill-r7) [inline plugins/okta/komand_okta/tasks/monitor_logs/task.py:187]: minor could remove this as this (`_filter_and_sort_recent_events`) doesn't exist and just mention it's used to calculate the from time?

**Review by joneill-r7 — COMMENTED** [review]
One small minro around the comment in the filter method, and just checking Okta doesn't enforce a time period of max x day in a lookup window? but rest lgtm!

**Review by jerichardson-r7 — APPROVED** [review]

File: plugins/okta/komand_okta/tasks/monitor_logs/task.py (hunk 2 above)
Comment (ablakley-r7) [reply plugins/okta/komand_okta/tasks/monitor_logs/task.py]: Also checked with a lookback of 5000 hours. It works ok.

**Pull request merged**